### ISSUE TITLE
BalenaApplicationNotFound for valid app id - preload cmd

### ISSUE THREAD

**Issue by nslocum** [balena-io/balena-cli#2063]
The `preload` cmd returns error `BalenaApplicationNotFound` when i pass an application id:

```shell
balena preload $IMAGE_FILENAME \
    --app $APP_ID \
    --commit current \
    --splash-image "$DIR/logo.png"
```

Returns error: 

`Application not found: 1234567 (BalenaApplicationNotFound)`

It works fine when I use the app name instead of its id:

```shell
balena preload $IMAGE_FILENAME \
    --app $APP_NAME \
    --commit current \
    --splash-image "$DIR/logo.png"
```

I get the app id using:

```shell
APP_ID=$(balena apps | grep " $APP_NAME " | awk '{ print $1 }')
```

This error really isn't an issue in my case, using the app name is actually easier. As an aside, I'd even prefer that the app ids be internal only and not exposed to any customers or customer APIs. By itself this isn't a serious security issue, but it needlessly exposes info that an attacker could easily leverage.

**Comment by pdcastro** [comment]
Fixed in CLI v12.33.1

**Comment by pdcastro** [comment]
> As an aside, I'd even prefer that the app ids be internal only and not exposed to any customers or customer APIs. By itself this isn't a serious security issue, but it needlessly exposes info that an attacker could easily leverage.

👍   This issue has been closed re preload command, but another related generic issue re app IDs remains open: #1843
